### PR TITLE
feat(client): add Long256 support

### LINKED ISSUE #3: Add support for Long256 type
https://questdb.io/docs/reference/api/ilp/columnset-types#long256

### PR DESCRIPTION
**Pull request by JammUtkarsh**
Closes #3

long256 support along with unit tests has been added go questDB client. 
Integration test will soon be added to be added to the system.

### CODE REVIEW

**Comment by puzpuzpuz** [comment]
Please add an integration test. It should be enough to add a test case or a few to the `TestE2EValidWrites` table test.

File: sender.go [hunk 1]
```diff
@@ -323,6 +323,30 @@ func (s *LineSender) Int64Column(name string, val int64) *LineSender {
...
+	s.lastErr = s.writeColumnName(name)
+	if s.lastErr != nil {
+		return s
+	}
+	s.buf.WriteByte('=')
+	s.lastErr = s.writeStrValue(val, false)
+	if s.lastErr != nil {
+		return s
+	}
+	// s.buf.WriteByte('i')
```
Comment (puzpuzpuz) [inline sender.go:345]: Looks like a leftover.

File: sender.go [hunk 2]
```diff
@@ -323,6 +323,30 @@ func (s *LineSender) Int64Column(name string, val int64) *LineSender {
 	return s
 }
 
+// Long256Column adds a 256-bit unsigned integer(long256) column
+// value to the ILP message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) Long256Column(name, val string) *LineSender {
```
Comment (puzpuzpuz) [inline sender.go:332]: nit: would be nice to accept `math/big.Int` instead of a raw string.

File: sender.go [hunk 3]
```diff
@@ -323,6 +323,30 @@ func (s *LineSender) Int64Column(name string, val int64) *LineSender {
 	return s
 }
 
+// Long256Column adds a 256-bit unsigned integer(long256) column
```
Comment (puzpuzpuz) [inline sender.go:326]: Expected `val` format should be documented.

File: sender_integration_test.go [hunk 4]
```diff
@@ -333,6 +336,26 @@ func TestE2EValidWrites(t *testing.T) {
 				Count: 1,
 			},
 		},
+		{
+			"single column long",
```
Comment (puzpuzpuz) [inline sender_integration_test.go:340]:
```suggestion
			"single column long256",
```

File: sender.go (hunk 2 above)
Comment (JammUtkarsh) [reply sender.go:332]: We tried the implementation with `math/big.Int` but there are a few unnecessary steps we need to perform on both the user's progam(like `main.go`)and in the go-questDB-client. 
Example:
`main.go` file
```go
    // rest of the code.
	hexaString := "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
	bigVal := new(big.Int)
	bigVal.SetString(hexaString, 16)
	err = sender.
		Table("trades").
		Symbol("name", "test_ilp1").
		Long256Column("value", bigVal).
		At(ctx, time.Now().UnixNano())
    // rest of the code.
```
& 
`sender.go` file
```go
func (s *LineSender) Long256Column(name string, val *big.Int) *LineSender {
    // rest of the code
    
    // adding the prefix and suffix.
	s.lastErr = s.writeStrValue("0x"+val.String()+"i", false)
    
    // rest of the code
}
```
Another limitation is that the `big.NewInt()` takes int64 which is also limited to 64 bits.
To solve this we also have `fmt.Sscan()` which takes number(int, long, float, etc)string and converts it into an any desired format.
Example:
```go
	i := new(big.Int)
	_, err := fmt.Sscan("115792089237316195423570985008687907853269984665640564039457584007913129639935", i)
    	if err != nil {
		log.Println("error scanning value:", err)
	} else {
		fmt.Println(i)
	}
```
Output: `115792089237316195423570985008687907853269984665640564039457584007913129639935`
Again, for sending `i` or output to the questDB we would need to perform addition steps of converting it into a hexadecimal and add prefix and postfix.

This is the reason, we implemented string. It offers a simple and straightforward solution.

Comment (puzpuzpuz) [reply sender.go:335]: There is [`SetBytes()`](https://pkg.go.dev/math/big#Int.SetBytes) method which can be used to initialize the integer without dealing with a string.

File: sender.go [hunk 5]
```diff
@@ -323,6 +323,32 @@ func (s *LineSender) Int64Column(name string, val int64) *LineSender {
 	return s
 }
 
+// Long256Column adds a 256-bit unsigned integer (long256) column
+// value to the ILP message.
+// 
+// val is pure unsigned hexadecimal string. Any attempt to set a string
```
Comment (puzpuzpuz) [inline sender.go:329]: Could you change the first sentence to something like `val should contain a hex encoded 256-bit unsigned integer value with "0x" prefix and "i" suffix.`?

**Review by puzpuzpuz — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
https://github.com/questdb/go-questdb-client/pull/7#discussion_r1057122341 can be addressed in a separate PR.

**Review by tris0laris — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
@JammUtkarsh many thanks for your contribution! Please let me know if you have plans to submit one more PR that would address https://github.com/questdb/go-questdb-client/pull/7#discussion_r1057122341

**Comment by JammUtkarsh** [comment]
Yes, @puzpuzpuz I would love to work on the comment mentioned. Please give me some time to get a better understanding of the topic and I shall resolve it soon.

**Comment by puzpuzpuz** [comment]
> Yes, @puzpuzpuz I would love to work on the comment mentioned. Please give me some time to get a better understanding of the topic and I shall resolve it soon.

Sure thing. Please let me know if you need any help. Look forward to your new contribution!

**Pull request merged**